components/brave_today/browser/suggestions_controller_unittest.cc

```diff
@@ -0,0 +1,254 @@
+    result[std::to_string(next_id++)] = std::move(publisher);
+  }
+  return result;
+}
+
+history::QueryResults MakeQueryResults(const std::vector<std::string>& urls) {
+  history::QueryResults result;
+  std::vector<history::URLResult> url_results;
+  for (const auto& url : urls) {
+    url_results.emplace_back(history::URLResult(GURL(url), base::Time::Now()));
+  }
+
+  result.SetURLResults(std::move(url_results));
+  return result;
+}
+}  // namespace
+
+class SuggestionsControllerTest : public testing::Test {
+ public:
+  SuggestionsControllerTest()
+      : api_request_helper_(TRAFFIC_ANNOTATION_FOR_TESTS,
+                            test_url_loader_factory_.GetSafeWeakWrapper()),
+        direct_feed_controller_(profile_.GetPrefs(), nullptr),
+        unsupported_publisher_migrator_(profile_.GetPrefs(),
+                                        &direct_feed_controller_,
+                                        &api_request_helper_),
+        publishers_controller_(profile_.GetPrefs(),
+                               &direct_feed_controller_,
+                               &unsupported_publisher_migrator_,
+                               &api_request_helper_),
+        suggestions_controller_(profile_.GetPrefs(),
+                                &publishers_controller_,
+                                &api_request_helper_,
+                                nullptr) {
+    SetLocale("en_US");
+  }
+  ~SuggestionsControllerTest() override = default;
+
+ protected:
+  std::vector<std::string> GetSuggestedPublisherIds(
+      const Publishers& publishers,
+      const history::QueryResults& history) {
+    return suggestions_controller_.GetSuggestedPublisherIdsWithHistory(
+        publishers, history);
+  }
+
+  void SetSimilarityMatrix(
+      SuggestionsController::PublisherSimilarities similarities) {
+    suggestions_controller_.similarities_ = std::move(similarities);
+  }
+
+  void SetLocale(const std::string& locale) {
+    suggestions_controller_.locale_ = locale;
+  }
+
+  content::BrowserTaskEnvironment browser_task_environment_;
+  TestingProfile profile_;
+  network::TestURLLoaderFactory test_url_loader_factory_;
+  api_request_helper::APIRequestHelper api_request_helper_;
+
+  DirectFeedController direct_feed_controller_;
+  UnsupportedPublisherMigrator unsupported_publisher_migrator_;
+  PublishersController publishers_controller_;
+  SuggestionsController suggestions_controller_;
+};
+
+TEST_F(SuggestionsControllerTest, VisitedSourcesAreSuggested) {
+  const auto& publishers = MakePublishers({
+      "https://example.com",
+      "https://bar.com",
+      "https://foo.com",
+  });
+  const auto& history = MakeQueryResults(
+      {"https://example.com", "https://foo.com", "https://example.com"});
+  auto suggestions = GetSuggestedPublisherIds(publishers, history);
+  EXPECT_EQ(2u, suggestions.size());
+  EXPECT_EQ("1", suggestions[0]);
+  EXPECT_EQ("3", suggestions[1]);
```

Comment:
nit: would be nice to have a comment for each EXPECT about why we're looking for these values, especially because we're passing in disconnected IDs generated by `MakePublishers`. Same for the other tests.